src/components/calculator/calculator.js

```diff
@@ -25,28 +26,31 @@ const buttons = [
   '=',
 ];
 
+const initialObject = { total: 0, next: null, operation: null };
 const Calculator = () => {
-  const [calculatorInputs, setCalculatorInputs] = useState('0');
+  // const [calculatorInputs, setCalculatorInputs] = useState('0');
+  const [obj, setObj] = useState(initialObject);
+
   const handleClick = (e) => {
     const { value } = e.target;
-    let newCalculatorInputs = calculatorInputs;
-    if (calculatorInputs[0] === '0') {
-      newCalculatorInputs = calculatorInputs.slice(1);
-      setCalculatorInputs(newCalculatorInputs);
-    }
-    if (value === 'AC') {
-      setCalculatorInputs('0');
-    } else {
-      setCalculatorInputs(newCalculatorInputs + value);
-    }
+
+    const result = calculate(obj, value);
+    setObj(result);
+    console.log(result);
```

Comment:
- [ ] You can use `console log`  when debugging but following the best practices, you are meant to remove this once you are done and not push it to your GitHub, I'll suggest you do so. :+1: